docs/how-to/cli-apps.rst

```diff
@@ -0,0 +1,157 @@
+
+.. code-block:: console
+
+    $ briefcase new
+
+When prompted to select a GUI framework, enter ``4`` to choose **Console**.
+This changes how Briefcase structures the app and configures the project.
+
+The most important change is that your ``pyproject.toml`` will include:
+
+.. code-block:: toml
+
+    console_app = true
+
+This flag informs Briefcase to treat the app as a terminal-based (non-GUI) application.
+Console apps print to the terminal directly, unlike GUI apps which log to the system
+console or display windows.
+
+Inspecting the File Structure
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Your generated project will have a layout similar to:
+
+.. code-block:: text
+
+    hello-cli/
+    ├── pyproject.toml
+    ├── src/
+    │   └── hello_cli/
+    │       ├── __main__.py
+    │       └── app.py
+    └── ...
+
+Note the ``app.py`` file contains a simple ``print("Hello, World.")`` statement by default.
+
+Running the App with Briefcase
+------------------------------
+
+Briefcase provides two ways to run your app: ``dev`` mode and ``run`` mode.
+You can pass command-line arguments to your app in both cases using ``--`` followed
+by your arguments.
+
+For example, running in dev mode:
+
+.. code-block:: console
+
+    $ briefcase dev -- John
+    ===========================================================================
+    Hello, John!
+
+And similarly after packaging:
+
+.. code-block:: console
+
+    $ briefcase run -- John
+    ===========================================================================
+    Hello, John!
+
+This is a key difference from GUI apps, which are usually launched without passing
+arguments and display a graphical window.
+
+Handling Command-line Arguments
+-------------------------------
+
+You can use any standard Python tool to parse CLI arguments. Here's a simple example
+using :any:`argparse`:
+
+.. code-block:: python
+
+    import argparse
+
+    def main():
+        parser = argparse.ArgumentParser(description="A simple CLI app.")
+        parser.add_argument("name", help="Your name")
+        args = parser.parse_args()
+
+        print(f"Hello, {args.name}!")
+
+Update your ``app.py`` file with this logic. You can now pass a name when running the app:
+
+.. code-block:: console
+
+    $ briefcase dev -- John
+    ===========================================================================
+    Hello, John!
+
+Packaging and Distribution
+---------------------------
+
+Once you've tested your app, you can package it for distribution using:
+
+.. code-block:: console
+
+    $ briefcase create
+    $ briefcase build
+    $ briefcase package
+
+For **macOS**, console apps are packaged as ``.pkg`` files, rather than the ``.app``
+bundles used for GUI apps. This is another major distinction.
+
+.. code-block:: console
+
+    $ briefcase package --adhoc-sign
+    ...
+    [hello-cli] Packaged dist/Hello CLI-0.0.1.pkg
+
+This ``.pkg`` file installs the app globally, and it can run from the terminal:
```

Comment:
It would be worth noting that the executable name is the app name, not the formal name. This is implied by the `hello-cli` example, but it's worth being explicit (and using `hello-cli` as an example).